src/amo/components/SponsoredAddonsShelf/index.js

```diff
@@ -52,12 +56,17 @@ export type InternalProps = {|
 export const formatDataForBeacon = ({
   data,
   key,
+  type,
 }: {|
   data: string,
   key: string,
+  type?: string,
```

Comment:
when will `type` be missing?  it's the key of the `event_data` dict/object where the data value came from.

Reply:
When we call `formatDataForBeacon` in order to send a beacon for `impression_url` (i.e., to `/shelves/sponsored/impression/`), we don't need a `type`.

Reply:
oh, it's used for impressions too.  I didn't catch that.